File: LeetCode Solutions/Linked List/Merge Two Sorted Lists.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */


struct ListNode* mergeTwoLists(struct ListNode* l1, struct ListNode* l2){
    struct ListNode *newNode = void ListNode(0);
    struct ListNode *requiredNode = newNode;
    while(l1 != NULL || l2 != NULL){
        if(l2 == NULL){
            requiredNode->next = l1;
            break;
        }
        if(l1 == NULL){
            requiredNode->next = l2;
            break;
        }
        if(l1->val > l2->val){
            requiredNode->next = void ListNode(l2->val);
            l2 = l2->next;
        } else{
            requiredNode->next = void ListNode(l1->val);
            l1 = l1->next;
        }
        requiredNode = requiredNode->next;
    }
    return newNode->next;
}

File: LeetCode Solutions/LeetCode 75 Study Plan to Ace Interviews/Level 1/Merge Two Sorted Lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        newnode = ListNode(0)
        requiredNode = newnode
        while(True):
            if not list1:
                requiredNode.next = list2
                break
            if not list2:
                requiredNode.next = list1
                break
            if list1.val > list2.val:
                requiredNode.next = ListNode(list2.val)
                list2 = list2.next
            else:
                requiredNode.next = ListNode(list1.val)
                list1 = list1.next
            requiredNode = requiredNode.next
        return newnode.next